src/Build.UnitTests/Instance/HostServices_Tests.cs

```diff
@@ -281,6 +295,20 @@ public void TestNonContraditcoryHostObjectAllowed_Any()
             Assert.Equal(NodeAffinity.InProc, hostServices.GetNodeAffinity("project"));
         }
 
+#if FEATURE_COM_INTEROP
+        /// <summary>
+        /// Test which ensures the remote host object cannot affect a project which has the Any affinity specifically set.
+        /// </summary>
+        [Fact]
+        public void TestReigsterRemoteHostObjectNoAffect_Any2()
+        {
+            HostServices hostServices = new HostServices();
+            hostServices.SetNodeAffinity("project", NodeAffinity.Any);
+            hostServices.RegisterHostObject("project", "target", "task", "moniker");
+            Assert.Equal(NodeAffinity.Any, hostServices.GetNodeAffinity("project"));
```

Comment:
We've been trying to move to Shouldly assertions for new code:

```suggestion
            hostServices.GetNodeAffinity("project").ShouldBe(NodeAffinity.Any);
```